api/src/main/java/org/eclipse/microprofile/metrics/MetricID.java

```diff
@@ -0,0 +1,236 @@
+    }
+
+    /**
+     * Returns the Metric name associated with this MetricID
+     *
+     * @return the Metric name associated with this MetricID
+     */
+    public String getName() {
+        return name;
+    }
+
+    /**
+     * Returns the underlying HashMap containing the tags.
+     *
+     * @return a {@link Map} of tags
+     */
+    public Map<String, String> getTags() {
+        return Collections.unmodifiableMap(tags);
+    }
+
+    /**
+     * Gets the list of tags as a single String in the format 'key="value",key2="value2",...'
+     *
+     * @return a String containing the tags
+     */
+    public String getTagsAsString() {
+        return this.tags.entrySet().stream().map(e -> e.getKey() + "=\"" + e.getValue() + "\"").collect(Collectors.joining(","));
+    }
+
+    /**
+     * Gets the list of tags as a list of {@link Tag} objects
+     *
+     * @return a a list of Tag objects
+     */
+    public List<Tag> getTagsAsList() {
+        List<Tag> list = new ArrayList<Tag>();
+        this.tags.entrySet().stream().forEach(entry -> list.add(new Tag(entry.getKey(), entry.getValue())));
+        return Collections.unmodifiableList(list);
+    }
+
+    /** {@inheritDoc} */
+    @Override
+    public boolean equals(Object o) {
+        if (this == o) {
+            return true;
+        }
+        if (!(o instanceof MetricID)) {
+            return false;
+        }
+        MetricID that = (MetricID) o;
+        return Objects.equals(this.name, that.getName()) && Objects.equals(this.tags, that.getTags());
+    }
+
+    /** {@inheritDoc} */
+    @Override
+    public int hashCode() {
+        return Objects.hash(name, tags);
+    }
+
+    /**
+     * Add multiple {@link Tag} objects
+     * This method will call {@link #addTag(Tag)} on each tag.
+     * If a key already exists then it will be overwritten.
+     *
+     * @param tagArray an array of {@link Tag} objects
+     */
+    private void addTags(Tag[] tagArray) {
+        if (tagArray == null || tagArray.length == 0) {
+            return;
+        }
+        Stream.of(tagArray).forEach(this::addTag);
+    }
+
+    /**
+     * Adds a singular {@link Tag} object into the MetricID.
+     * If the tag key/name already exists then it will be overwritten with
+     * the new value.
+     *
+     * @param tag the {@link Tag} object
+     */
+    private void addTag(Tag tag) {
+        if (tag == null || tag.getTagName() == null || tag.getTagValue() == null) {
+            return;
+        }
+        tags.put(tag.getTagName(), tag.getTagValue());
+    }
+
+    /**
+     * Parses the global tags retrieved from environment variable {@code MP_METRICS_TAGS}.
+     *
+     * @param globalTags the string of global tags retrieved from MP_METRICS_TAGS
+     * @throws IllegalArgumentException if the global tags list does not adhere to
+     *             the appropriate format.
+     */
+    private void parseGlobalTags(String globalTags) throws IllegalArgumentException {
```

Comment:
Should tags with an empty value be allowed? Currently this won't be able to parse `prop=` and fail with an `ArrayIndexOutOfBoundsException`, either this should be accepted or a more user-friendly exception should be thrown